mezzanine/utils/custom_content_types.py

```diff
@@ -0,0 +1,152 @@
+"""
+Mixin classes for models that can be subclassed to create custom types.
+In order to use them:
+
+- Inherit model from CustomContentTypes.
+- Inherit that model's ModelAdmin from CustomContentTypesAdmin.
+- Allow selection of custom models in the change_list, in the spirit of
+templates/admin/pages/page/change_list.html.
+"""
+from copy import deepcopy
+
+from django.apps import apps
+from django.core.urlresolvers import NoReverseMatch
+from django.db import models
+from django.http import HttpResponseRedirect
+from django.shortcuts import get_object_or_404
+
+from mezzanine.utils.models import base_concrete_model
+from mezzanine.utils.urls import admin_url
+
+
+class CustomContentTypes(models.Model):
```

Comment:
> Not a fan of "Custom" in the name, I'd prefer something like "ContentTyped". Thoughts?

I didn't like the name either but couldn't think of anything better. "ContentTyped" it is.